### PR TITLE
tests: Improve and document ability to filter/select tests to run

### PR DESCRIPTION
**Pull request by grahamwhaley**
Add a `make help` to the tests Makefile to list all the test targets that are run by default, and thus are also available as direct test target rules.
Enhance the ginkgo SKIP/FOCUS in the Makefile a touch by adding those to the functional target rule.
Document how to use the `make help`, bats files and ginkgo skip/focus for directed test running.

### CODE REVIEW

**Comment by grahamwhaley** [comment]
So, my PR grew a touch from 'document ginkgo focus and skip', and I dropped a couple of related niceties in, like `make help`. I can split this to another PR if need be, but I tried to make it all look and feel cohesive as a single PR ;-)

File: Makefile [hunk 1]
```diff
@@ -171,6 +172,10 @@ $(INSTALL_TARGETS): install-%: .ci/install_%.sh
 list-install-targets:
 	@echo $(INSTALL_TARGETS) | tr " " "\n"
 
+help:
+	@echo Subsets of the tests can be run using the follwoing specific make targets:
```
Comment (jodh-intel) [inline Makefile]: Typo: "following"

File: README.md [hunk 2]
```diff
@@ -218,6 +222,60 @@ the docker integration tests, run the following:
...
+```
+$ make help
+```
+
+### Running subsets of tests
+
+Individual tests or subsets of tests can be selected to be run. The method of
+test selection depends on which type of test framework the test is written
+with. Most of the Kata Containers test suites are written
+using either [BATS](https://github.com/sstephenson/bats) files or with
```
Comment (jodh-intel) [inline README.md]: Nit: Although "BATS" makes sense to me, the official name appears to be "Bats".

File: README.md [hunk 3]
```diff
@@ -218,6 +222,60 @@ the docker integration tests, run the following:
...
+necessary components. An example of how a `BATS` test is run from the `Makefile`
+looks like:
+
+```makefile
+kubernetes:
+        bash -f .ci/install_bats.sh
+        bash -f integration/kubernetes/run_kubernetes_tests.sh
+```
+
+#### Running `Ginkgo` based tests
```
Comment (jodh-intel) [inline README.md]: Nit: You don't need the backticks here (and elsewhere) when referring to the name of the project as the spell-checker will understand this term.

File: README.md [hunk 4]
```diff
@@ -218,6 +222,60 @@ the docker integration tests, run the following:
...
+
+`Ginkgo` supports selecting, filtering and excluding tests to be run using command
+line parameters. The tests repository `Makefile` supports passing the environment
+variables `FOCUS` and `SKIP` through to the `ginkgo` command in most cases. Check
+the `Makefile` specifics before you use this functionality.
+
+`ginkgo` accepts [`Golang` `regexp`](https://golang.org/pkg/regexp/) regular expressions
```
Comment (jodh-intel) [inline README.md]: Again, you can write "Golang" and "regex" here as they will be accepted by the spell-checker.

**Comment by grahamwhaley** [comment]
And I forgot the # on the fixes! so travis is mumbling.
@jodh-intel - thx. I hovered abound bats/BATS, and `quoting` product names or not - but thought I'd let the doc bot... err, you parse that for me ;-) I'll update and push...

**Comment by grahamwhaley** [comment]
/test
respun and pushed.

**Pull request merged**